Snippets/Snippets_5/Features/FeatureConfiguration.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace Snippets5.Features
+{
+    using System;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.Pipeline;
+
+    class MyMagicFeature : Feature
+    {
+
+#region FeatureConfiguration
+        public MyMagicFeature()
+        {
+            EnableByDefault();
+
+            DependsOn<Sagas>();
+
+            Prerequisite(c => c.Settings.EndpointName().StartsWith("Magical"), "This feature only works on magical endpoints");
+        }
+#endregion
+
+#region FeatureSetup
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            // retrieve some global settings
+            var supervised = context.Settings.Get<bool>("SupervisorActive");
+
+            if (!supervised)
+            {
+                // register a more performant cleanup process with the container
+                context.Container.ConfigureComponent<UnsafeCleanupComponent>(DependencyLifecycle.InstancePerCall);
+            }
+
+            // add a custom behavior to the message processing pipeline which does our cleanup.
+            context.Pipeline.Register<CleanupBehaviorRegistration>();
+        }
+#endregion
+
+        void ConfigureFeatures()
+        {
+#region EnableDisableFeatures
+            var configuration = new BusConfiguration();
+            
+            // enable Saga feature since our custom feature relies on it
+            configuration.EnableFeature<Sagas>();
+
+            // this is not required if the feature uses EnableByDefault()
+            configuration.EnableFeature<MyMagicFeature>();
```

Comment:
shouldnt the enabling of saga be owned by the endpoint config code not a feature? if i was in a real large apps dev team and i saw this i would recommend instead the feature throw an exception if it detects sagas no being enable. this way it would avoid the unintended side effect of someone using the feature and sagas unexpectantly enabling